rs/state_manager/src/lib.rs

```diff
@@ -2791,11 +2805,30 @@ impl StateManager for StateManagerImpl {
                           "Copying checkpoint {} with root hash {:?} under new height {}",
                           checkpoint_height, root_hash, height);
 
+                    match self.state_layout.checkpoint_verification_status(checkpoint_height) {
+                        Ok(true) => {}
+                        Ok(false) => {
+                            warn!(self.log,
+                                "Unverified checkpoint @{} is not expected to be found in states metadata.",
```

Comment:
I think this message is misleading, especially in the future where unverified checkpoints might live longer. The error is that the entry in the states metadata has a root hash (because we found it via `find_checkpoint_by_root_hash`) and is unverified. Unverified without a root hash wouldn't really be a problem.

Reply:
You are right. I wrote the comment based on that currently states metadata only contains entries of verified checkpoints.

In the future where unverified checkpoints might live longer and states metadata can contain entries of unverified checkpoints. The error could be a unverified checkpoint has a root hash or it does not hide the root hash. It depends on how we block publishing the manifest.

I prefer not to going into details here. Maybe just generally say "Unverified checkpoint @{} cannot be used to cloned to a new checkpoint height" .